corehq/apps/reports/standard/cases/case_data.py

```diff
@@ -449,11 +449,7 @@ def case_xml(request, domain, case_id):
 @require_GET
 def case_property_names(request, domain, case_id):
     case = safely_get_case(request, domain, case_id)
-    all_property_names = set(CaseProperty.objects.filter(
-        case_type__domain=domain,
-        case_type__name=case.type,
-        deprecated=False,
-    ).values_list('name', flat=True))
+    all_property_names = set(case.dynamic_case_properties())
     all_property_names = all_property_names.difference(KNOWN_CASE_PROPERTIES) | {"case_name"}
```

Comment:
With this change, we no longer need to remove known case properties either right? Though would still need to add back `case_name`.

Reply:
I can never remember if it's possible to get any of those special properties into `case.case_json`. We don't want to allow people to edit them so this should be fine (even if it's not necessary most of the time), right?